erigon-lib/state/domain_shared.go

```diff
@@ -1434,9 +1465,6 @@ func _decodeTxBlockNums(v []byte) (txNum, blockNum uint64) {
 // LatestCommitmentState searches for last encoded state for CommitmentContext.
 // Found value does not become current state.
 func (sdc *SharedDomainsCommitmentContext) LatestCommitmentState() (blockNum, txNum uint64, state []byte, err error) {
-	if sdc.patriciaTrie.Variant() != commitment.VariantHexPatriciaTrie {
```

Comment:
|| != commitment.VariantHexPatriciaTrieReader`, eg if binary is given. check needed to happen anyway. for now LatestCommitmentState is supported only for hex patricia but with time support will expand.

Reply:
but different types could be added but commitment state saving for them is not supported